### ISSUE TITLE
Install of PrestaShop 1.7.8-rc.1 is not possible in CLI mode

### ISSUE THREAD

**Issue by PrestaEdit** [PrestaShop/PrestaShop#25959]
#### Describe the bug

Install of PrestaShop 1.7.8-rc.1 is not possible in CLI mode.
I used this command
`php index_cli.php --language=fr --country=fr --domain=prestashop.com --base_uri=rc1780 --db_server=10.1.32.2 --db_user=root --db_name=1780cli --db_password=xxx --firstname=Demo --lastname=Prestashop --email=demo@prestashop.com --password=prestashop_demo --db_create=1 --newsletter=0 --send_email=0 --enable_ssl=1`

#### Expected behavior

Install in CLI mode work as before.

#### Steps to Reproduce

Make a composer create-project prestashop/prestashop:1.7.8-rc.1 and try to use the index_cli in order to install PrestaShop.

You will see this logs:

```
TypeError: Argument 1 passed to PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint::shop() must be of the type int, null given, called in /src/Adapter/Configuration.php on line 422 and defined in /src/Core/Domain/Shop/ValueObject/ShopConstraint.php:62
Stack trace:
 
0 /src/Adapter/Configuration.php(422): PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint::shop()
1 /src/Adapter/Configuration.php(106): PrestaShop\PrestaShop\Adapter\Configuration->buildShopConstraintFromContext()
2 /src/Core/Addon/Theme/ThemeRepository.php(74): PrestaShop\PrestaShop\Adapter\Configuration->get()
3 /classes/shop/Shop.php(495): PrestaShop\PrestaShop\Core\Addon\Theme\ThemeRepository->getInstanceByName()
4 /classes/shop/Shop.php(137): ShopCore->setTheme()
5 /install-dev/controllers/console/process.php(71): ShopCore->__construct()
6 /install-dev/controllers/console/process.php(227): InstallControllerConsoleProcess->initializeContext()
7 /install-dev/controllers/console/process.php(138): InstallControllerConsoleProcess->processPopulateDatabase()
8 /install-dev/classes/controllerConsole.php(117): InstallControllerConsoleProcess->process()
9 /install-dev/index_cli.php(64): InstallControllerConsole::execute()
```

**Screenshots**

If applicable, add screenshots or screen recordings to help explain your problem.

#### Additional information

* PrestaShop version: 1.7.8-rc.1
* PHP version: N/A

**Comment by hibatallahAouadni** [comment]
Hello @PrestaEdit 

I didn't manage to reproduce your issue with PS1780_rc1, see the attached screenshot below:

![image](https://user-images.githubusercontent.com/23308427/134331145-b9d7c43c-7134-48f8-9202-2fd5f06f068d.png)

Please check and feedback.
Thanks!

**Comment by PrestaEdit** [comment]
There is my complete console cmd:

`php index_cli.php --domain=my.domain.tld --base_uri=/foldername/1.7.8.0-rc.1 --language=fr --db_name=mydbname --db_user=mydbuser --db_password=mydbpassword --db_create=1 --name=myshopname --firstname=Presta --lastname=Module --password=mypassword --email=email@tld.com --newsletter=0 --send_email=0 --enable_ssl=1`

**Comment by PrestaEdit** [comment]
Issue seems to be inside the: --base_uri=/foldername/1.7.8.0-rc.1/

**Comment by khouloudbelguith** [comment]
Hi @PrestaEdit,

I manage to reproduce the issue with PS1780rc
I used this command
` php index_cli.php --language=fr --country=fr --domain=prestashop.com --base_uri=rc1780 --db_server=10.1.32.2 --db_user=root --db_name=1780cli --db_password=xxx --firstname=Demo --lastname=Prestashop --email=demo@prestashop.com --password=prestashop_demo --db_create=1 --newsletter=0 --send_email=0 --enable_ssl=1`
I have this error 

```
TypeError: Argument 1 passed to PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint::shop() must be of the type integer, null given, called in /var/www/html/rc1780/src/Adapter/Configuration.php on line 422 and defined in /var/www/html/rc1780/src/Core/Domain/Shop/ValueObject/ShopConstraint.php:62
Stack trace:
#0 /var/www/html/rc1780/src/Adapter/Configuration.php(422): PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopConstraint::shop(NULL)
#1 /var/www/html/rc1780/src/Adapter/Configuration.php(106): PrestaShop\PrestaShop\Adapter\Configuration->buildShopConstraintFromContext()
#2 /var/www/html/rc1780/src/Core/Addon/Theme/ThemeRepository.php(74): PrestaShop\PrestaShop\Adapter\Configuration->get('_PS_ALL_THEMES_...')
#3 /var/www/html/rc1780/classes/shop/Shop.php(495): PrestaShop\PrestaShop\Core\Addon\Theme\ThemeRepository->getInstanceByName('classic')
#4 /var/www/html/rc1780/classes/shop/Shop.php(137): ShopCore->setTheme()
#5 /var/www/html/rc1780/install/controllers/console/process.php(71): ShopCore->__construct(1)
#6 /var/www/html/rc1780/install/controllers/console/process.php(227): InstallControllerConsoleProcess->initializeContext()
#7 /var/www/html/rc1780/install/controllers/console/process.php(138): InstallControllerConsoleProcess->processPopulateDatabase()
#8 /var/www/html/rc1780/install/classes/controllerConsole.php(117): InstallControllerConsoleProcess->process()
#9 /var/www/html/rc1780/install/index_cli.php(64): InstallControllerConsole::execute(17, Array)
```
With PS1778 => ok
![image](https://user-images.githubusercontent.com/16067358/134687370-a20affc3-7628-421b-b7f8-db1e683e0bd4.png)

I'll add this to the debug backlog so that it's fixed.

Please be aware that due to the high number of reported issues, some bugs might take a very long time to be resolved. If this issue is important to you and you cannot wait for it to be fixed on the project's own time, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you. 

If you manage to get the problem fixed, then please contribute the fix back to the project as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/). Remember that the more people contribute, the better PrestaShop becomes for everyone.

Thanks!

**Comment by matks** [comment]
> Issue seems to be inside the: --base_uri=/foldername/1.7.8.0-rc.1/

Looks like we dont test enough the "install prestashop in a subfolder" usecase

**Comment by atomiix** [comment]
Fixed by #25982

**Comment by alberto-garcia-cocomarch** [comment]
We are on version 1.7.8.2

public static function shopGroup(?int $shopGroupId, bool $isStrict = false): self

Still is not showing the proper fix?
[2022-01-27 09:50:08] php.INFO: User Deprecated: Not specifying the optional ShopConstraint parameter is deprecated since version 1.7.8.0 {"exception":"[object] (ErrorException(code: 0): User Deprecated: Not specifying the optional ShopConstraint parameter is deprecated since version 1.7.8.0 at /home2/dracoco/public_html/upgrade.dracocomarch.com/src/Adapter/Configuration.php:418)"} []

Please advise
Thanks
